discussion/src/main/java/org/openedx/discussion/presentation/threads/DiscussionThreadsViewModel.kt

```diff
@@ -159,20 +160,16 @@ class DiscussionThreadsViewModel(
     }
 
     fun sortThreads(orderBy: String) {
-        if (lastOrderBy != orderBy) {
-            lastOrderBy = orderBy
-            threadsList.clear()
-            nextPage = 1
-        }
+        lastOrderBy = orderBy
+        threadsList.clear()
+        nextPage = 1
         loadThreads()
```

Comment:
Since we are refactoring the code, I believe it’s better to retain the condition and place `loadThreads()` inside it to ensure the app doesn't reload threads for the same order when necessary.

Reply:
But in that case, nothing will happen when the user clicks the same filter or sort. 🤔